gast_to_code.py

```diff
@@ -1,45 +1,51 @@
-
-# general_helpers
-def value_to_str(val):
-    if type(val) == str:
-        return '"' + val + '"'
-    else:
-        return str(val)
-
-def body(gast_list, out_lang):
+"""
+Helper for lists of gast
+Default is to put comma and space btwn each stringified gast
+    i.e. list_helper({str_gast}, {str_gast}, out_lang) --> str, str
+Can specify different btwn string with third parameter
+    i.e. list_helper({str_gast}, {str_gast}, out_lang, "**") --> str**str
+"""
+def list_helper(gast_list, out_lang, btwn_str = ", "):
     out = ""
     for gast in gast_list:
         out += gast_router(gast, out_lang)
-        out += "\n"
-
-    return out[:-1] # remove \nS
-
-def list_to_csv_str(l):
-    s = ""
-    for i in l:
-        if type(i) == str:
-            i = '"' + i + '"'
-        s += str(i) + ", "
+        out += btwn_str
     
-    return s[:-2] # remove last comma and space
-
+    num_btwn_chars = len(btwn_str)
+    return out[:-num_btwn_chars] # remove \nS
+        
 # py_specific_helpers
 def py_logStatement(gast):
-    arg_string = list_to_csv_str(gast["args"])
+    arg_string = gast_router(gast["args"],"py")
     return "print(" + arg_string + ")"
 
 def py_varAssign(gast):
-    value = value_to_str(gast["varValue"])
+    value = gast_router(gast["varValue"], "py")
     return gast["varId"] + " = " + value
 
 # js_specific_helpers
 def js_logStatement(gast):
-    arg_string = list_to_csv_str(gast["args"])
+    arg_string = gast_router(gast["args"],"js")
     return "console.log(" + arg_string + ")"
 
 def js_varAssign(gast):
-    value = value_to_str(gast["varValue"])
-    return "const " + gast["varId"] + " = " + value
+    kind = gast["kind"]
+    varId = gast["varId"]
+    varValue = gast_router(gast["varValue"], "js")
+    
+    return kind + " " + varId + " = " + varValue
+
+def py_bool(gast):
+    if gast["value"] == 1:
+        return "True"
+    else:
+        return "False"
+
+def js_bool(gast):
+    if gast["value"] == 1:
```

Comment:
Any reason we're using numbers to represent booleans in the gAST? JSON supports booleans natively.

Reply:
Not really- we just made that decision together since we kept on getting confused with capitalization. Also when converting back to code, 0 often is false and 1 is true which we liked. Do we need to change?